### PR TITLE
update curl examples

### PR DESCRIPTION
**Pull request by zbayoff**
### What does this PR do?
Update curl examples to fall back to global DD api and app keys and content type if not specifically provided for the endpoint.

### Preview link
http://docs-staging.datadoghq.com/zach/fix-curl/api/v1/monitors/#monitors-group-search

check that all curl examples now have default
```
-H "Content-Type: application/json" \
-H "DD-API-KEY: ${DD_CLIENT_API_KEY}" \
-H "DD-APPLICATION-KEY: ${DD_CLIENT_APP_KEY}"
```

### CODE REVIEW

File: layouts/api/list.html [hunk 1]
```diff
@@ -418,14 +418,21 @@ <h3 class="mb-2">Code Example</h3>
...
 <span class="o">-H</span> <span class="s1">"{{ $authSchema.name }}: <span class="n">${ {{ index $authSchema "x-env-name" }} }</span>"</span>
 {{- end -}}
 {{- end -}}
-{{- if $context.action.requestBody -}} \
+{{- else -}}
+{{ $securitySchemes := $d.components.securitySchemes }}
+<span class="o">-H</span> <span class="s1">"{{ $securitySchemes.apiKeyAuth.name }}: <span class="n">${ {{ index $securitySchemes.apiKeyAuth "x-env-name" }} }</span>"</span> \
+<span class="o">-H</span> <span class="s1">"{{ $securitySchemes.appKeyAuth.name }}: <span class="n">${ {{ index $securitySchemes.appKeyAuth "x-env-name" }} }</span>"</span> 
+{{- end -}}
+{{- if $context.action.requestBody -}}&nbsp;\
```
Comment (jirikuncar) [inline layouts/api/list.html:435]: it's inside `<code>`
```suggestion
{{- if $context.action.requestBody -}} \
```

File: layouts/api/list.html [hunk 2]
```diff
@@ -418,14 +418,21 @@ <h3 class="mb-2">Code Example</h3>
...
 {{- range $key, $val := . -}}
 {{ $authSchema := (index $d.components.securitySchemes $key) }}
 <span class="o">-H</span> <span class="s1">"{{ $authSchema.name }}: <span class="n">${ {{ index $authSchema "x-env-name" }} }</span>"</span>
 {{- end -}}
 {{- end -}}
-{{- if $context.action.requestBody -}} \
+{{- else -}}
+{{ $securitySchemes := $d.components.securitySchemes }}
+<span class="o">-H</span> <span class="s1">"{{ $securitySchemes.apiKeyAuth.name }}: <span class="n">${ {{ index $securitySchemes.apiKeyAuth "x-env-name" }} }</span>"</span> \
+<span class="o">-H</span> <span class="s1">"{{ $securitySchemes.appKeyAuth.name }}: <span class="n">${ {{ index $securitySchemes.appKeyAuth "x-env-name" }} }</span>"</span>
```
Comment (jirikuncar) [inline layouts/api/list.html:433]:
```suggestion
<span class="o">-H</span> <span class="s1">"{{ $securitySchemes.apiKeyAuth.name }}: <span class="n">${{{ index $securitySchemes.apiKeyAuth "x-env-name" }}}</span>"</span> \
<span class="o">-H</span> <span class="s1">"{{ $securitySchemes.appKeyAuth.name }}: <span class="n">${{{ index $securitySchemes.appKeyAuth "x-env-name" }}}</span>"</span> 
```

File: layouts/api/list.html [hunk 3]
```diff
@@ -418,14 +418,21 @@ <h3 class="mb-2">Code Example</h3>
...
 {{- end -}}
-{{- end -}}
-{{- range $context.action.security -}}
+{{ else }}
+-H "Content-Type: application/json" \{{- end -}}
+{{- with $context.action.security -}}
+{{- range . -}}
 {{- range $key, $val := . -}}
 {{ $authSchema := (index $d.components.securitySchemes $key) }}
 <span class="o">-H</span> <span class="s1">"{{ $authSchema.name }}: <span class="n">${ {{ index $authSchema "x-env-name" }} }</span>"</span>
```
Comment (jirikuncar) [inline layouts/api/list.html:427]:
```suggestion
<span class="o">-H</span> <span class="s1">"{{ $authSchema.name }}: <span class="n">${{{ index $authSchema "x-env-name" }}}</span>"</span>
```

File: layouts/api/list.html [hunk 4]
```diff
@@ -418,14 +418,21 @@ <h3 class="mb-2">Code Example</h3>
 {{range $key, $val := . }}
 -H "Content-Type: {{ $key }}" \{{ end }}
 {{- end -}}
-{{- end -}}
-{{- range $context.action.security -}}
+{{ else }}
+-H "Content-Type: application/json" \{{- end -}}
```
Comment (jirikuncar) [inline layouts/api/list.html:422]:
```suggestion
-H "Content-Type: application/json" \
{{- end -}}
```

File: layouts/api/list.html [hunk 5]
```diff
@@ -418,14 +418,21 @@ <h3 class="mb-2">Code Example</h3>
 {{range $key, $val := . }}
 -H "Content-Type: {{ $key }}" \{{ end }}
```
Comment (jirikuncar) [inline layouts/api/list.html:419]:
```suggestion
-H "Content-Type: {{ $key }}" \
{{- end -}}
```

File: layouts/api/list.html (hunk 1 above)
Comment (zbayoff) [reply layouts/api/list.html:435]: yeah i need to use &nbsp; a 'space' isn't rendering. &nsbp; is not rendering in the output though is it? Not even when you copy right?

Comment (jirikuncar) [reply layouts/api/list.html:437]: `{{- if $context.action.requestBody }} \` should work

File: layouts/api/list.html [hunk 6]
```diff
@@ -416,16 +416,25 @@ <h3 class="mb-2">Code Example</h3>
 {{- with $context.action.requestBody -}}
 {{- with .content -}}
 {{range $key, $val := . }}
--H "Content-Type: {{ $key }}" \{{ end }}
+-H "Content-Type: {{ $key }}" \
```
Comment (jirikuncar) [inline layouts/api/list.html:419]:
```suggestion
<span class="o">-H</span> <span class="s1">"Content-Type: {{ $key }}"</span> \
```

File: layouts/api/list.html [hunk 7]
```diff
@@ -416,16 +416,25 @@ <h3 class="mb-2">Code Example</h3>
...
 {{- end -}}
 {{- end -}}
-{{- range $context.action.security -}}
+{{ else }}
+-H "Content-Type: application/json" \
```
Comment (jirikuncar) [inline layouts/api/list.html:423]:
```suggestion
<span class="o">-H</span> <span class="s1">"Content-Type: application/json"</span> \
```

File: layouts/api/list.html [hunk 8]
```diff
@@ -426,13 +441,15 @@ <h3 class="mb-2">Code Example</h3>
...
+{{- if eq $authSchema.in "header" -}}
 <span class="o">-H</span> <span class="s1">"{{ $authSchema.name }}: <span class="n">${ {{- index $authSchema "x-env-name" -}} }</span>"</span>
 {{- end -}}
 {{- end -}}
+{{- end -}}
 {{- else -}}
 {{ $securitySchemes := $d.components.securitySchemes }}
 <span class="o">-H</span> <span class="s1">"{{ $securitySchemes.apiKeyAuth.name }}: <span class="n">${ {{- index $securitySchemes.apiKeyAuth "x-env-name" -}} }</span>"</span> \
-<span class="o">-H</span> <span class="s1">"{{ $securitySchemes.appKeyAuth.name }}: <span class="n">${ {{- index $securitySchemes.appKeyAuth "x-env-name" -}} }</span>"</span> 
+<span class="o">-H</span> <span class="s1">"{{ $securitySchemes.appKeyAuth.name }}: <span class="n">${ {{- index $securitySchemes.appKeyAuth "x-env-name" -}} }</span>"</span>
```
Comment (zbayoff) [inline layouts/api/list.html:452]:
```suggestion
<span class="o">-H</span> <span class="s1">"{{ $securitySchemes.appKeyAuth.name }}: <span class="n">${ {{- index $securitySchemes.appKeyAuth "x-env-name" -}} }</span>"</span> \
```

File: layouts/api/list.html [hunk 9]
```diff
@@ -426,13 +441,15 @@ <h3 class="mb-2">Code Example</h3>
...
 {{- end -}}
 {{- if $context.action.requestBody -}}&nbsp;\
```
Comment (zbayoff) [inline layouts/api/list.html:454]:
```suggestion
{{- if $context.action.requestBody -}}&nbsp;
```

**Pull request merged**